app/models/concerns/geo_concerns/metadata_extraction_helper.rb

```diff
@@ -2,16 +2,16 @@ module GeoConcerns
   module MetadataExtractionHelper
     # Extracts properties from the constitutent external metadata file
     # @return [Hash]
-    def extract_metadata
+    def extract_metadata(id)
       return {} if metadata_files.blank?
-      # TODO: Does not support multiple external metadata files
-      raise NotImplementedError if metadata_files.length > 1
-      metadata_files.first.extract_metadata
+      metadata_files.each do |metadata_file|
```

Comment:
In this case, the order isn't relevant. It could be written more succinctly with something like:
```
metadata_file = metadata_files.find { |f| f.id == if }
metadata_file.extract_metadata if metadata_file
```
The method is simply search through a collection of metadata files (ordered or not) to find one with an id matching one passed in with the `should_populate_metadata` parameter. The logic in this helper module needs to be updated to accept id params whether or not we are using an ajax interface. Perhaps that param should have a different name. `populate_metadata_with` or something. But I think I get your point. We should be careful not to couple our business logic with our UI too tightly.